modules/imaging_browser/php/imaging_browser.class.inc

```diff
@@ -284,117 +98,119 @@ END";
+        // that are desired to go into imaging browser table, based on
+        // array values rather than keys (hence the array_flip), then flip
+        // the resulting array back to revert it to a key/value (i.e.
+        // acquisition protocol ID/scan type) combination.
+        $config             = \NDB_Config::singleton();
+        $toTable_scan_types = $config->getSetting('tblScanTypes');
+        $scan_id_types      = array_flip(
+            array_intersect_key(
+                array_flip($all_scan_types),
+                array_flip($toTable_scan_types)
+            )
         );
 
-        $this->fieldOptions = [
-                               'sites'         => $siteList,
-                               'projects'      => $list_of_projects,
-                               'visitQCStatus' => $visitQCStatus,
-                               'sequenceTypes' => $scan_types,
-                               'pendingNew'    => $pending,
-                               'configLabels'  => $scan_id_types,
-                              ];
+        return [
+                'sites'         => $siteList,
+                'projects'      => $list_of_projects,
+                'visitQCStatus' => $visitQCStatus,
+                'sequenceTypes' => $all_scan_types,
+                'pendingNew'    => $pending,
+                'configLabels'  => $scan_id_types,
+               ];
 
-        $this->tpl_data['backURL'] = $_SERVER['REQUEST_URI'];
     }
+
     /**
-     * Converts this menu filter to an array of the form
-     *
-     * Headers => (string array)
-     * Data => (array of arrays of row data)
+     * Gets the data source for this menu filter.
      *
-     * @note overloaded function
-     *         Overloading this method to create a list of sessionID that
-     *         will be used for the Navigation Links in  the viewSession
-     *         page.
+     * @return \LORIS\Data\Provisioner
+     */
+    public function getBaseDataProvisioner() : \LORIS\Data\Provisioner
+    {
+        return new ImagingBrowserRowProvisioner();
+    }
+
+    /**
+     * Return a data provisioner of the same type as BaseDataProvisioner, with
+     * additional LORIS filters applied.
      *
-     * @return array
+     * @return \LORIS\Data\Provisioner a provisioner with specific filters added
      */
-    function toArray()
+    public function getDataProvisionerWithFilters(): \LORIS\Data\Provisioner
     {
-        $data  = parent::toArray();
-        $index = array_search('SessionID', $data['Headers']);
-        if ($index !== false) {
-            $_SESSION['State']->setProperty(
-                'mriSessionsListed',
-                array_column($data['Data'], $index)
-            );
+
+        $provisioner = parent::getDataProvisionerWithFilters();
+
+        $user = \User::singleton();
+
+        // UserSiteMatch for imaging browser specific permissions
+        if ($user->hasPermission("imaging_browser_view_allsites") == false) {
```

Comment:
I think this might be easier to read/review if you used "&&" to structure it more like a truth table:

ie
```
if($user->hasPermission(A) && $user->hasPermission(B)) {
    // description of what that means
} else if($user->hasPermission(A) && !user->hasPermission(B)) {
    // description of what that means
} else if(!$user->hasPermission(A) && user->hasPermission(B)) {
    // description of what that means
} else {
   // has neither, description of what that means
}
```

Reply:
Here is the logic I came up with for `ProjectImagesRowProvisioner`. 
```
        // If a user can't view images from all sites check if they can view images
        // from its own sites. If not, limit images to phantoms.
        if (!$user->hasPermission('imaging_browser_view_allsites')) {
            if ($user->hasPermission('imaging_browser_view_site')) {
                $new = $new->filter(
                    new \LORIS\Data\Filters\UserSiteMatch()
                );
            } else {
                $new = $new->filter(
                    new \LORIS\Data\Filters\OnlyPhantoms()
                );
            }
        }

        // If a user can't view phantoms from all sites, check if they can view
        // phantoms from its own sites. If not, filter out phantoms.
        if (!$user->hasPermission('imaging_browser_phantom_allsites')) {
            if ($user->hasPermission('imaging_browser_phantom_ownsite')) {
                $new = $new->filter(
                    new \LORIS\Data\Filters\UserSiteMatch()
                );
            } else {
                $new = $new->filter(
                    new \LORIS\Data\Filters\NoPhantoms()
                );
            }
        }
        return $new;
```
I your case, replace `$new` by `$provisioner`.
I hope it helps.

There is still a "weak" point where a user that do not have `imaging_browser_view_allsites` but have `imaging_browser_view_site` will get only images and phantom from his own sites regarless of if he have `imaging_browser_phantom_allsites`. That is because `UserSiteMatch` applies to every rows, not just on images or phantom. This is yet an other argument to have a specific menu filter for phantoms.

Reply:
right. Good catch. Will add it in the next commit.

I totally agree that there should be two different menu filters for phantoms and candidates. Would be much easier to maintain. But I guess it is behind the scope of this PR though...

Reply:
If you think that's a good idea to split it and make things easier, maybe it would be a good idea to do that first and block this until it's done?

Reply:
I think it would be easier to merge this one and then work on splitting this it into two menu filters. The work to convert the imaging browser module to the data framework is done. 

All that would be left is splitting this into two menu filters for next round. Seems a lot easier to first merge this and then work on the split.